File: mobile/app/src/main/java/com/resuera/eventhive/features/settings/SettingsApi.kt
package com.resuera.eventhive.features.settings

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.PUT

interface SettingsApi {

    @PUT("/api/users/profile")
    fun updateProfile(@Body body: Map<String, String>): Call<Map<String, Any>>

    @PUT("/api/users/password")
    fun changePassword(@Body body: Map<String, String>): Call<Map<String, String>>
}

File: web/eventhive/src/features/settings/settingsApi.ts
import { apiFetch } from '../../shared/api/client'

export interface UpdateProfileRequest {
  firstname: string
  lastname: string
  email: string
}

export interface ChangePasswordRequest {
  currentPassword: string
  newPassword: string
}

export const settingsApi = {
  updateProfile: (data: UpdateProfileRequest) =>
    apiFetch<any>('/api/users/profile', {
      method: 'PUT',
      body: data,
    }),

  changePassword: (data: ChangePasswordRequest) =>
    apiFetch<any>('/api/users/password', {
      method: 'PUT',
      body: data,
    }),

  uploadProfilePic: (image: File) => {
    const formData = new FormData()
    formData.append('image', image)
    return apiFetch<{ profilePicUrl: string }>('/api/users/profile-pic', {
      method: 'POST',
      body: formData,
    })
  },
}